nuitka/nodes/IterationHandles.py

```diff
@@ -0,0 +1,310 @@
+#     Copyright 2019, Kay Hayen, mailto:kay.hayen@gmail.com
+#
+#     Part of "Nuitka", an optimizing Python compiler that is compatible and
+#     integrates with CPython, but also works on its own.
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+#
+""" Node for Iteration Handles.
+
+"""
+
+from abc import abstractmethod
+
+from nuitka.__past__ import getMetaClassBase
+
+
+class IterationHandleBase(getMetaClassBase("IterationHandle")):
+    """Base class for Iteration Handles."""
+
+    @abstractmethod
+    def __repr__(self):
+        """Returns a printable representation of the IterationHandle
+        and it's children object."""
+        return
+
+    @abstractmethod
+    def getNextValueExpression(self):
+        """Abstract method to get next iteration value."""
+        return
```

Comment:
Doesn't pylint hate those useless returns? Puting the docstring is enough, can leave out the return statement, unless this would be a default implementation actually called by base class manually.

Reply:
Nope. Initially, I used `pass` in place of `return`, pylint2 was not happy with it.

Reply:
Yes, `pass` is also not needed.

You see, doc strings in Python are statements. A string as a statement has not much effect, and therefore, Python allows its parser to special case those strings as first statements.

But that is why `def f(): ""` is a complete function definition. Python injects a `return` at the end of each function body automatically. And the `pass` is not needed, because, well there is already a statement, that is why it complained. The fact that PyLint does not complain about the return doesn't convince me of it being good. Just remove them, and PyLint and I will both be happy.

As far as blocks go, you can do these:

```
while 1:
   ""
```

Not that are any good, just to illustrate what is happening. PyLint is probably going to hate that one though, because it's not a doc string. Statement without effect is what I would expect to be warned about.

So documenting your functions has the extra bonus of allowing to remove the `pass` that you initially likely had there.